TakingITGlobal/commit2act
# [TIG-724] Separate page components structure for signup flow

## Ticket(s)

* [TIG-724](https://app.clickup.com/t/9009201449/TIG-724) (focused sub-task)
* [TIG-663](https://app.clickup.com/t/9009201449/TIG-663) (parent task)

## Description

This PR sets up the pattern for component names and rendering within `PublicRoutes` for sub-pages of the Signup flow. For the scope of this, only the first page (`InfoPrimary`) has been created, with the others to follow in [the next sub-task](https://app.clickup.com/t/9009201449/TIG-725).  

We're also introducing the page-level `useSignupPageRoute` hook to pull each page's routing, step, and progress info from the routing array using `useLocation` at time of render. 

## Testing

* Pull down and install the branch
* Run tests (`npm run test`) and confirm they all pass
* Run Storybook (`npm run storybook`) and wait for it to open in-browser
* Confirm the Views > SignupContainer stories are rendering 
* Confirm the Pages > Sign Up > Step 1 - InfoPrimary story is rendering
* Run the app (`npm run start`) and wait for it to open in-browser
* Navigate to the first Sign Up page (`/account/create`) and confirm the heading "Great!" from the `InfoPrimary` component renders
* Hit the "Next" link and confirm the fallback "Signup Page" heading is rendered

## Screenshots

**`InfoPrimary` page rendered in Storybook:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/f5cb15cc-ec33-41a1-85c5-75be82f1c350)

**Initial page component rendering with the first Sign Up route within the local app:

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/99eae4dc-698b-41b4-a9df-b055884efcaf)

**Fallback heading rendered for a later step in the flow:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/22abe78b-7835-450a-adf8-d2c3bc50c723)
